app/gui/view/component-browser/component-group/src/icon.rs

```diff
@@ -1,156 +1,108 @@
-    (middle + top + bottom).into()
-}
+/// This constant exists for development purposes only, and is published for debug scene.
+/// Due to a rendering error, shapes appear too big when the camera is zoomed in very closely.
+/// (Documented here: https://github.com/enso-org/ide/issues/1698)
+/// In the user interface, this is not a big problem, since icon are usually shown at lower zoom
+/// levels. But it is a problem during development of icon when it becomes necessary to inspect
+/// them closely. In those situations, one can apply `.shrink(0.35)` to shapes to compensate for the
+/// bug and make them appear at the correct size while the camera is zoomed in. But that work-around
+/// will make them appear too thin on the default zoom level.
+///
+/// To make it easy to turn this shrinking on and off before and after working on icon, we define
+/// the constant `SHRINK_AMOUNT` and apply `.shrink(SHRINK_AMOUNT.px())` to all icon. In every
+/// commit, `SHRINK_AMOUNT` should be set to 0.0 to make icon look best in the user interface. But
+/// during work on the icon, it can temporarily be set to 0.35.
+pub const SHRINK_AMOUNT: f32 = 0.0;
 
-/// An arc around the origin. `outer_radius` determines the distance from the origin to the outer
-/// edge of the arc, `stroke_width` the width of the arc. The arc starts at `start_angle`, relative
-/// to the origin and ends at `end_angle`. The ends are flat not rounded, as in [`RoundedArc`].
-fn arc(outer_radius: f32, stroke_width: f32, start_angle: f32, end_angle: f32) -> AnyShape {
-    let circle = Circle(outer_radius.px()) - Circle((outer_radius - stroke_width).px());
-    let inner_angle = (end_angle - start_angle).rem_euclid(2.0 * PI);
-    let mid_angle = start_angle + inner_angle / 2.0;
-    let angle = PlaneAngleFast(inner_angle.radians()).rotate(mid_angle.radians());
-    // The implementation of `PlaneAngleFast` adds 0.5 px to the actual distance to avoid artifacts
-    // in corner cases. We apply `grow` to compensate for that and get the shape that we really
-    // want.
-    let angle = angle.grow(0.5.px());
-    (circle * angle).into()
-}
 
-/// The shape of a table, given by a grid with size `columns` x `rows`. The stroke width is 1.0 and
-/// the cell size 4.0. The origin is at the lower left corner.
-fn table(columns: i32, rows: i32) -> AnyShape {
-    const STROKE_WIDTH: f32 = 1.0;
-    const CELL_SIZE: f32 = 4.0;
-
-    let width = columns as f32 * CELL_SIZE + STROKE_WIDTH;
-    let height = rows as f32 * CELL_SIZE + STROKE_WIDTH;
-    let bounds =
-        Rect((width.px(), height.px())).translate(((width / 2.0).px(), (height / 2.0).px()));
-    let grid = grid(STROKE_WIDTH, CELL_SIZE);
-    (grid * bounds).into()
+
+// ==============
+// === Errors ===
+// ==============
+
+/// Error occuring when we try parse string being an invalid icon name to icon Id.
+#[derive(Clone, Debug, Fail)]
+#[fail(display = "Unknown icon {}.", name)]
```

Comment:
Use `"Unknown icon '{}'."` instead. It would be easier to read if someone passed an empty string for example.